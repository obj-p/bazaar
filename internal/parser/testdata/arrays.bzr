package arrays

data ArrayFields {
    empty [int] = []
    singleValue [string] = ["hello, value!"]
    trailingComma [bool] = [true,]
    optionalValue [int?] = [null]
    optionalArray [int]?
}

func Arrays() {
    var emptyArray = []
    var array = [1, 2, 3,]
}
