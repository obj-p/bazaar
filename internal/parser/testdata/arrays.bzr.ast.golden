&parser.Bazaar{Package: &parser.PackageDecl{Domain: []string{"arrays"}}, Decls: []*parser.TopLevelDecl{{Data: &parser.DataDecl{Name: "ArrayFields", Members: []*parser.MemberDecl{{Field: &parser.FieldDecl{Name: "empty", Type: &parser.TypeDecl{Array: &parser.ArrayTypeDecl{Value: parser.TypeDecl{Value: &"int"}}}, Default: &parser.Expr{Unary: &parser.UnaryExpr{Primary: &parser.PrimaryExpr{Literal: &parser.Literal{Array: &parser.ArrayLiteral{}}}}}}}, {Field: &parser.FieldDecl{Name: "singleValue", Type: &parser.TypeDecl{Array: &parser.ArrayTypeDecl{Value: parser.TypeDecl{Value: &"string"}}}, Default: &parser.Expr{Unary: &parser.UnaryExpr{Primary: &parser.PrimaryExpr{Literal: &parser.Literal{Array: &parser.ArrayLiteral{Values: []*parser.Expr{{Unary: &parser.UnaryExpr{Primary: &parser.PrimaryExpr{Literal: &parser.Literal{String: &parser.String{Fragments: []*parser.StringFragment{{Text: &"hello, value!"}}}}}}}}}}}}}}}, {Field: &parser.FieldDecl{Name: "trailingComma", Type: &parser.TypeDecl{Array: &parser.ArrayTypeDecl{Value: parser.TypeDecl{Value: &"bool"}}}, Default: &parser.Expr{Unary: &parser.UnaryExpr{Primary: &parser.PrimaryExpr{Literal: &parser.Literal{Array: &parser.ArrayLiteral{Values: []*parser.Expr{{Unary: &parser.UnaryExpr{Primary: &parser.PrimaryExpr{Literal: &parser.Literal{Bool: &parser.Bool(true)}}}}}}}}}}}}, {Field: &parser.FieldDecl{Name: "optionalValue", Type: &parser.TypeDecl{Array: &parser.ArrayTypeDecl{Value: parser.TypeDecl{Value: &"int", Optional: true}}}, Default: &parser.Expr{Unary: &parser.UnaryExpr{Primary: &parser.PrimaryExpr{Literal: &parser.Literal{Array: &parser.ArrayLiteral{Values: []*parser.Expr{{Unary: &parser.UnaryExpr{Primary: &parser.PrimaryExpr{Literal: &parser.Literal{Null: true}}}}}}}}}}}}, {Field: &parser.FieldDecl{Name: "optionalArray", Type: &parser.TypeDecl{Array: &parser.ArrayTypeDecl{Value: parser.TypeDecl{Value: &"int"}}, Optional: true}}}}}}}}
